### PR TITLE
Enable more C++ warnings

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @EikanWang @jgong5 @wenzhe-nrv @sanchitintel @mingfeima @XiaobingSuper @ashokei @jingxu10

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks!

File: aten/src/ATen/native/quantized/library.cpp [hunk 1]
```diff
@@ -1,13 +1,5 @@
...
-
-int register_linear_params();
-
-template <int kSpatialDim = 2>
-int register_conv_params();
-
-extern template int register_conv_params<2>();
-extern template int register_conv_params<3>();
-int register_embedding_params();
+#include <aten/src/ATen/native/quantized/cpu/fbgemm_utils.h>
```
Comment (malfet) [inline aten/src/ATen/native/quantized/library.cpp:2]: This type of change is bad, as it add dependency between native and native/cpu. We probably should move fbgemm_utils.h to native if we want to do something like that

Comment (cyyever) [reply aten/src/ATen/native/quantized/library.cpp]: Follow the suggestion by moving the declarations to aten/src/ATen/native/quantized/library.h

**Pull request closed without merging**